docs/sphinx/developer/arch/supported/aarch64.rst

```diff
@@ -211,9 +211,13 @@ Loads and stores
 
 In addition to an execution behaviour, memory instructions also require a new entry in the address generation behaviour table found in ``src/lib/arch/aarch64/Instruction_address.cc``. These entries are responsible for describing the method used to generate the addresses that these instructions will read from or write to.
 
-Address generation is expected to generate one or more instances of ``MemoryAddressTarget``, containing an address and the number of bytes to access. The same variables described above (``operands``, ``metadata``) are available to use to generate these addresses.
+Address generation is expected to generate one or more instances of ``MemoryAccessTarget``, containing an address and the number of bytes to access. The same variables described above (``operands``, ``metadata``) are available to use to generate these addresses.
 
-Once the addresses have been generated, they should be supplied in a vector to the ``setMemoryAddresses`` helper function.
+Once the addresses have been generated, they should be supplied in a vector to the ``setMemoryAddresses`` helper function. 
+
+For loads, data can be read from the ``memoryData`` vector in ``Instruction_execute.cc``, with each index holding a ``RegisterValue`` for a corresponding ``MemoryAccessTarget``. For stores, a ``RegisterValue`` must be placed in each index of the ``memoryData`` vector, again one per ``MemoryAccessTarget`` generated.
+
+Concerning vector loads and stores (i.e. NEON, SVE, or SME), an effort should be made to create as few ``MemoryAccessTarget``'s as possible, with the exception of gather/scatter instructions where one ``MemoryAccessTarget`` should be created per vector element.
```

Comment:
I'm not sure of the best way to get my point across or a good replacement, but I think something like this would be better. "To best match modelled hardware, contiguous Load and Store instructions used one ``MemoryAccessTarget`` per destination/source register respectively whilst non-contiguous variants use multiple."

Reply:
Saying just one MemoryAccessTarget would be inaccurate right? given that it all depends on the predicate for SVE loads/stores? 
This is only true for NEON right? Even then we sometimes generate more than one MemoryAccessTarget, but the generated ones target a larger address range

Reply:
We could divide the description into two separate sentences i.e one of SVE/SME (as they are predicated) and one for NEON.
For SVE/SME you could say 

> An effort must be made to merge contiguous active elements as a single load/store by using predicate information. This has been done in helper functions `sve_merge_store_data` and `generatePredicatedContiguousAddressBlocks`. Depending upon the SVE/SME instruction to be implemented either these helpers could be used or similar logic be implemented.

For NEON we can say what @jj16791 wrote:

> To best match modelled hardware, contiguous Load and Store instructions used one MemoryAccessTarget per destination/source register

Reply:
I think a combination of the two could work. We are merging all of them to better match hardware, but a more verbose explanation can be done for SVE/SME for sure. 

I would change the SVE/SME one though to say "into a single load/store by evaluating the predicate information." And to also put in which of `Instruction_Address` and `Instruction_execute` these helper functions are used.

Also "Depending upon the SVE/SME instruction to be implemented, these helpers can be used or similar logic can be implemented if they are not suitable."